### ISSUE TITLE
On HyperV the podman machine creation fails if you set more memory than the one available

### ISSUE THREAD

**Issue by lstocchi** [containers/podman-desktop#6814]
### Bug description

Openin this issue as i faced this earlier.
When you try to create a podman machine on HyperV, and you set more memory that the one that is available the creation fails.
the memory slider should be adjusted with the maximum value one can set

### Operating system

Windows 11

### Installation Method

None

### Version

next (development version)

### Steps to reproduce

1. try to create a machine by adding maximum memory. It should fail.

**Comment by lstocchi** [comment]
@jeffmaury I'm not sure if we made a test about it with the new implementation. Worth checking. I can also try when switching on Windows if you don't check in the meantime

**Comment by jeffmaury** [comment]
To be verified

**Comment by rujutashinde** [comment]
Qe it to verify

**Comment by odockal** [comment]
It is still an issue.

**Comment by rujutashinde** [comment]
@odockal  after it was verified it was still and issue , we missed out on reassigning it to someone to take a look for this sprint/1.20.0 release. I believe this would need to move to `1.21.0`

**Comment by odockal** [comment]
Yes, I forgot to unassign myself. Please go ahead and move to next milestone.

**Comment by simonrey1** [comment]
@rujutashinde thanks for assigning me! I don't have a Windows machine set up here, and this seems to be a Windows-specific issue. I do have one at home that I could potentially use, but it would be much more straightforward if someone who already has a Windows development environment could take a look. Let me know what you think is best.

**Comment by jeffmaury** [comment]
I self assigned this issue as I have a Windows machine. But I'm not sure what is the solution as the memory parameter is already bounded by the host memory so should we add a new bound ? Pinging @odockal for comments ?

**Comment by odockal** [comment]
@jeffmaury I am not sure about implications of such change. I though that hyperv allows to set amount of memory to use for machine when creating one, in contradiction of wsl which is using value based on system configuration or config file...

**Comment by jeffmaury** [comment]
> [@jeffmaury](https://github.com/jeffmaury) I am not sure about implications of such change. I though that hyperv allows to set amount of memory to use for machine when creating one, in contradiction of wsl which is using value based on system configuration or config file...

Yes when configuring an HyperV machine the max memory is set to the physical memory available on the host but based on your host running applications the creation may fail. I would recommend to close it

**Comment by odockal** [comment]
I thin kthat the point of the issue was to not make user overly happy with possibility of setting more memory than his system offers :D
So only thing that could be done here is to change the upper value of the slider to what is being set in the slider? If that is easy fix, let's do it, if not, we can close.

**Comment by odockal** [comment]
Note: I think the problem might be that on my machine (32 GB Ram) I can set 33.5 GB memory on the slider.

**Comment by odockal** [comment]
Setting a memory over actually available mem on the system looks like this: 
<img width="983" height="927" alt="Image" src="https://github.com/user-attachments/assets/71716a26-ac73-4c70-9774-19953a1aad6a" />

**Comment by odockal** [comment]
But it looks the same when one chooses ie. 30 GB of Ram on running system,  but this is expected since that much memory is not possible to dedicate to a vm. I would just fix the max. number one can set on slider.

**Comment by jeffmaury** [comment]
> But it looks the same when one chooses ie. 30 GB of Ram on running system, but this is expected since that much memory is not possible to dedicate to a vm. I would just fix the max. number one can set on slider.

For me the slider value is correct as the unit is GB so if you have 32GiB on your machine the 33.5GB is correct. Closing then feel free to re open

**Comment by odockal** [comment]
@jeffmaury Still does not look right to me: I have (16 GB) of memory looking into About this PC, and PD allows me to set 16.5 GB, which are the same unit. Looking through the `systeminfo`: `Total Physical Memory:     16,046 MB`

<img width="1005" height="591" alt="Image" src="https://github.com/user-attachments/assets/2236ce62-6582-43c6-b2ae-d7ee58200af5" />

**Comment by odockal** [comment]
Well, well... some says that windows' GB notation actually mean GiB, but microsoft does not seem to update it in any way, although the conversion between GiB to GB is: 16 GiB = 17.179869184 GBm which also does not explains why we see 16.5 GB on slider ?!
Trying to get precise info about installed memory and correct units is not easy job on windows.

I encountered also this information, which is very close to what we see. 

<img width="259" height="115" alt="Image" src="https://github.com/user-attachments/assets/a2e96302-a94f-47df-a8d4-07ea49c656dc" />

This one is close to what we see in PD. In any way, trying to create a machine with 15.5 or 16.5 GB on machine with 16 GB throws the same error not actually tied to PD using number over allowed range. 

Keeping this issue closed is likely ok.
